src/game/eop.js

```diff
@@ -116,321 +109,324 @@ export const ElevationOfPrivilege = Game({
-        selectedThreat: "",
+    onEnd: onTurnEnd,
+    stages: {
+      threats: {
+        moves: {
+          addOrUpdateThreat,
+          deleteThreat,
+          pass,
+          selectDiagram,
+          selectComponent,
+          selectThreat,
+          toggleModal,
+          toggleModalUpdate,
+          updateThreat
+        },
       }
     },
-    selectThreat: (G, ctx, id) => {
-      // if the player has passed, they shouldn't be able to select threat
-      let passed = [...G.passed];
-      if (passed.includes(ctx.playerID)) {
-        return INVALID_MOVE;
-      }
+  },
+  
+  moves: {
+    draw,
+    selectDiagram,
+    selectComponent,
+    selectThreat,
+  },
+};
 
-      return {
-        ...G,
-        selectedThreat: id,
-      }
+function toggleModal(G, ctx) {
+  // if the player has passed, they shouldn't be able to toggle the modal
+  let passed = [...G.passed];
+  if (passed.includes(ctx.playerID) || (G.threat.modal && G.threat.owner !== ctx.playerID)) {
+    return INVALID_MOVE;
+  }
+  let card = getDealtCard(G);
+  return {
+    ...G,
+    threat: {
+      ...G.threat,
+      modal: !G.threat.modal,
+      new: true,
+      owner: ctx.playerID,
+      type: G.suit,
+      id: uuidv4(),
+      title: "",
+      severity: "Medium",
+      description: getThreatDescription(card),
+      mitigation: "",
     },
-    pass: (G, ctx, id) => {
-      let passed = [...G.passed];
-      
-      if (!passed.includes(ctx.playerID)) {
-        passed.push(ctx.playerID);
-      }
+  }
+}
 
-      return {
-        ...G,
-        passed,
-      }
+function toggleModalUpdate(G, ctx, threat) {
+  // if the player has passed, they shouldn't be able to toggle the modal
+  let passed = [...G.passed];
+  if (passed.includes(ctx.playerID) || (threat.owner !== ctx.playerID)) {
+    return INVALID_MOVE;
+  }
+
+  return {
+    ...G,
+    threat: {
+      ...G.threat,
+      modal: !G.threat.modal,
+      new: false,
+      id: threat.id,
+      owner: ctx.playerID,
+      title: threat.title,
+      type: threat.type,
+      severity: threat.severity,
+      description: threat.description,
+      mitigation: threat.mitigation,
     },
-    deleteThreat: (G, ctx, threat) => {
-      // if the player has passed, they shouldn't be able to toggle the modal
-      let passed = [...G.passed];
-      if (passed.includes(ctx.playerID) || (threat.owner !== ctx.playerID)) {
-        return INVALID_MOVE;
-      }
+  }
+}
 
-      let scores = [...G.scores];
-      scores[ctx.playerID]--;
+function updateThreat(G, ctx, field, value) {
+  return {
+    ...G,
+    threat: {
+      ...G.threat,
+      [field]: value,
+    }
+  }
+}
 
-      let identifiedThreats = _.cloneDeep(G.identifiedThreats);
-      delete identifiedThreats[G.selectedDiagram][G.selectedComponent][threat.id];
+function selectDiagram(G, ctx, id) {
+  // if the player has passed, they shouldn't be able to select diagrams
+  let passed = [...G.passed];
```

Comment:
You are using this pattern quite often. Maybe it deserves to the defined as its own function?
```js
function hasPlayerPassed(G, ctx) {
   return G.passed.includes(ctx.playerID);
}
```